### ISSUE TITLE
Is it possible to password protect root directory but not it's subfolders?

### ISSUE THREAD

**Issue by parnexcodes** [lyc8503/onedrive-cf-index-ng#3]
I would like to password protect my root folder so user can't see all my files.

**Comment by lyc8503** [comment]
Have you tried to set `protectedRoutes` to `["/"]` and add a `.password` file in the root directory?

**Comment by parnexcodes** [comment]
> Have you tried to set `protectedRoutes` to `["/"]` and add a `.password` file in the root directory?

Yes but it password protects the folders inside it too.

**Comment by lyc8503** [comment]
I think that's the expected behaviour, setting `/` as protected will protect all your files with a password, what do you expect to do?

**Comment by parnexcodes** [comment]
> I think that's the expected behaviour, setting `/` as protected will protect all your files with a password, what do you expect to do?

I want to share folders inside the root but not let the users see the whole index.

**Comment by lyc8503** [comment]
If users can't see the index, how can they know what directories are there and access the files? only via direct links?

This feature haven't been implemented yet.

Or maybe I didn't get you correct? Could you please describe your use case by giving a specific example?

**Comment by parnexcodes** [comment]
I want to share `Supernatural.Academy.S01.1080p.AMZN.WEB-DL.DD+5.1.H.264-playWEB` folder with users but I don't want to give them the permission to see the whole index.

They should be able to see only this folder and root folder should be password protected. This way they can't view/download all the files from my index.

**Comment by lyc8503** [comment]
Yes... This feature isn't implemented and probably not planned since it may cause confusion and difficulty in setup. Maybe you can reorganize your folder structure?

**Comment by alan16742** [comment]
> I want to share `XXX.XXX.S01.1080p.AMZN.WEB-DL.DD+5.1.H.264-XXX` folder with users but I don't want to give them the permission to see the whole index.
> 
> They should be able to see only this folder and root folder should be password protected. This way they can't view/download all the files from my index.
I don't know if you can use it now. Another onedrive index project `onemanager` is to turn on the `no password to download files` function, Turn on this function and know the link to directly download files other than password files. (Please ignore the machine translation error)

**Comment by lyc8503** [comment]
> > I want to share `XXX.XXX.S01.1080p.AMZN.WEB-DL.DD+5.1.H.264-XXX` folder with users but I don't want to give them the permission to see the whole index.
> > They should be able to see only this folder and root folder should be password protected. This way they can't view/download all the files from my index.
> 
> I don't know if you can use it now. Another onedrive index project `onemanager` is to turn on the `no password to download files` function, Turn on this function and know the link to directly download files other than password files. (Please ignore the machine translation error)

As I mentioned in https://github.com/lyc8503/onedrive-cf-index-ng/issues/8#issuecomment-1872424009, you should probably not rely on URL itself to keep your files private. Also, the issue itself is trying to share a subfolder without password protection, which is different from the `no password to download files` feature you're talking about.
